### ISSUE TITLE
BUG: `Real.toRational` hangs on `pow` function 

### ISSUE THREAD

**Issue by hmf** [non/spire#491]
Trying to use the `Real` to calculate an exponential using the latest stable version (0.10.1). When I create the type no problem. Its when the `toRational` is executed, whether directly or not, that it hangs.
Examples (all those in comments will hang):
```scala
    //val r2 = Real(2).fpow(Real(3))
    //val r2 = Real.e.fpow(Real(3))
    //val r2 = Real(2).pow(3.0)
    //val r2 = Real(2.0).pow(3.0)
    //val r2 = Real(2).fpow(3)
    val r2 = Real(2).pow(3)

    val s = r2.toRational // hangs here
 ```

The power function seems to work only when the exponent is an integer. I tracked this down to the class:
```scala
case class Inexact(f: Int => SafeLong) 
```
When the `apply`is invoked it enters the `None` (second) case. Then it execute `f(p)` and `Inexact.apply` is called again. The looping seems to start here.

This is a show stopper for me. Any way I can get around this?

**Comment by non** [comment]
@hmf Ouch! I'll investigate now. Thanks for reporting.

**Comment by hmf** [comment]
@non  I have tried to search a little more and it seems to lead to Rational@90: 
```scala 
def limitToInt: Rational = limitTo(BigInt(Int.MaxValue)) 
```
I now wonder if this is not the same issue as #393

**Comment by non** [comment]
OK, I see what is happening. It is an easy bug to fix.

As a work-around you can break apart a rational number and apply the `.pow` and `.nroot` calls separately. I'll try to get a fix released ASAP.

**Comment by non** [comment]
Yeah, it's definitely related to a problem with `.limitToInt`. In the case we care about it is easy to fix -- I'll have to see if it fixes that bug too.

**Comment by hmf** [comment]
Thanks for the suggested workaround. However I am working with `Real`. So I guess I have to convert to `Rational` first.  If so I think that won't do. The `Real.toRational` is what's getting me into trouble.  :-(

**Comment by non** [comment]
I'll try to get a release out with a fix for this ASAP.

**Comment by hmf** [comment]
Thank you very much.
